### ISSUE TITLE
Authentication Works In Local Dev Mode But Not In Production (AWS EC2)

### ISSUE THREAD

**Issue by Jiansen** [Shopify/shopify-app-template-remix#529]
# Issue summary

We build an App based on the Remix template and the Billing App.

The OAuth and Billing API works fine when we tested it locally (using `yarn dev` and build a cloudflare tunnel.
 
However, when we host the server on AWS EC2 and start using `npm run setup && PORT=8888 npm run start`
1. the admin page of the test store loaded. i.e. https://admin.shopify.com/store/quickstart-b74cf95b/apps/site-chat-shop-extension    where [quickstart-b74cf95b](https://admin.shopify.com/store/quickstart-b74cf95b/apps/site-chat-shop-extension) is the name of the test store.
2. Authentication always failed, neither redirected to the login page (if open the host home page URL in a new tab and, outside the Admin panel, it will redirect me to the login page)

For reference

1. The followings are code snippet from the `app/routes/_index/route.jsx`.

```
export const loader = async ({ request }) => {
  try{
    console.log('home page, auth check:');
    const { billing } = await authenticate.admin(request);
    console.log('home page, billing check:', billing);
    // some other code
  } catch (error) {
    console.log('home page, billing check error')
    console.log('home page, billing check error details:', error)
    console.log('home page, billing check error apiKey:', process.env.SHOPIFY_API_KEY || "")
    return json({
      apiKey: process.env.SHOPIFY_API_KEY || "",
    });
 }
 ```
 
  2. the log shows that authentication was attempted but failed.

![image](https://github.com/Shopify/shopify-app-template-remix/assets/1225761/b6846f6e-fd6b-45ba-9cc2-c45ef847f779)

3. Browser console shows that auth/billing returns `401` error

- `@shopify/shopify-app-remix` version: "^2.1.0",
- Node version: v21.5.0
- Yarn version: 1.22.21
- Operating system: local - MacOS; production: Ubuntu 12.04

**Comment by matteodepalo** [comment]
Hi @Jiansen, thank you for opening this issue. Could you provide a more detailed log of the request that happens in production including all the headers?

**Comment by Jiansen** [comment]
Thanks @matteodepalo ,


Although the logs suggest an authentication-related issue, I discovered that the actual cause is the access scopes being out of sync between the `.env` file (used in production deployment) and the `shopify.app.toml` (since local testing didn't check the `.env` file).

 

If anyone encounters a similar issue, consider:
- Check the scopes in both the `shopify.app.toml` file and the `.env` file.
- We've created an **npm** package: _shopify-scopes-sync_ (available on [npm](https://www.npmjs.com/package/shopify-scopes-sync) and [GitHub](https://github.com/Jiansen/shopify-scopes-sync)) that automatically syncs the scopes in the .env file as specified in shopify.app.toml.

 

Hope this helps.
